### ISSUE TITLE
`configure` defaults to setting `COQPREFIX=local`

### ISSUE THREAD

**Issue by peterlefanulumsdaine** [coq/coq#14468]
#### Description of the problem

I’m trying to build+install current `master` (ea359584), on Mac OS Big Sur 11.4.  When I call `./configure` without options and accept all its defaults, then `make world` builds fine, but `make install` doesn’t install anything, giving the following output:

```
pll coq % make install
/Library/Developer/CommandLineTools/usr/bin/make --warn-undefined-variable --no-builtin-rules -f Makefile.build install
DUNE      sources
Prefix not set in configure, nothing to install.
```

Looking at `Makefile.install`, it sounds like this is expected when `configure` has been given the deprecated `-local` option; and indeed my `config/Makefile` includes `COQPREFIX=local`.  But I didn’t explicitly issue `-local`; I ran `configure` without options, and accepted its default locations, which all use `/usr/local` as prefix (e.g. `Where should I install the Coq binaries [/usr/local/bin]? ` […] `the Coq binaries will be copied in /usr/local/bin`).

If I explicitly call `configure  -prefix '/usr/local'`, then it correctly writes `COQPREFIX=/usr/local` in `config/Makefile`, and `make install` works as expected.

So it looks like `configure` is defaulting to setting `COQPREFIX=local`, and it seems like this probably isn’t intended behaviour?  At least, this is a change of behaviour — in the past (e.g. v8.13) the `-prefix` didn’t need to be explicitly specified, and the instructions in `INSTALL.make.md` don’t mention that it should be.

For comparison, v8.13 installs fine with `./configure` defauls.  It looks like this may be to do with the changes in `https://github.com/coq/coq/pull/14176`, but I haven’t managed to precisely confirm that, since when I try to build the versions immediately before/after that, I get build failures with seemingly unrelated errors.

#### Coq Version

As mentioned above, this issue occurs with ea359584 (current `master`).  I’m on Mac OS Big Sur 11.4, using OCaml 4.12.0.

**Comment by Zimmi48** [comment]
> my `config/Makefile` includes `COQPREFIX=local`

I can reproduce on Linux (which makes sense as I don't see how this could have been macOS-specific). That's indeed not intended and a serious bug. Thanks for reporting!

**Comment by ejgallego** [comment]
Indeed the logic in that part is hard to follow, this is a bug as it should ask you about the prefix in that case.

I tried to fix it in #14189 but didn't manage to advance enough in time, I'll prepare a hotfix for now.

**Comment by ejgallego** [comment]
I have made some progress on this, I recall now why I kept the local case, was for those doing `-profile devel` + `make install`, I guess we can just have `make install` work in that case too.

**Comment by Alizter** [comment]
hmmm this was closed by #14515 but I'm not so sure without some testing like we discussed in the call.

**Comment by ejgallego** [comment]
AFAICT this should be really fixed by #14515 , tho not in the cleanest way.
